front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -100,16 +118,41 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
-  const [tmpSelectedTrain, setTmpSelectedTrain] = useState(selectedTrainScheduleId);
-
-  useEffect(() => {
-    setTmpSelectedTrain(selectedTrainScheduleId);
-  }, [selectedTrainScheduleId]);
+  const projectedTrainsAndOccurrences = useMemo(
+    () =>
+      projectPathTrainResult.flatMap((train) => {
+        if (isOccurrence(train.id) && train.paced) {
```

Comment:
- early return if the train is a trainSchedule
- you can create a function `isTrainSpaceTimeDataWithPaced` so we don't have any typing issues

Reply:
Make an early return here trigger a linter error https://www.typescriptlang.org/tsconfig/#noImplicitReturns